packages/preview-middleware-client/test/unit/cpe/control-data.test.ts

```diff
@@ -197,9 +192,16 @@ describe('controlData', () => {
 
     test('buildControlData', async () => {
         // act
-        const result = await buildControlData(control as any, controlOverlay as any);
+        const result = await buildControlData(control as any, mockOverlay as any);
 
         // assert
         expect(result).toMatchSnapshot();
+
+        expect(control.getMetadata).toBeCalledWith();
```

Comment:
If you don't check the calling parameter, just use `toBeCalled` please.

Now about the test itself: is it relevant that `control.getMetadata` has been called? Same for the other checks.
The question you should ask is what is `buildControlData` supposed to do and how can I verify it. Without looking at the function itself, I would assume that we need to ensure that the result is as expected, so I would assume that it would be a good idea to validate that the the provided control and overlay content matches the expectations of the result.

Reply:
adapted.